transformer_engine/pytorch/fp8.py

```diff
@@ -638,7 +646,7 @@ def _compute_scaling_factor(
             fp8_max,
             recipe.margin,
         )
-    return recipe.scaling_factor_compute_algo(amax, scale, fp8_max, recipe)
+    return recipe.scaling_factor_compute_algo(amax, scale, fp8_max, recipe.margin)
```

Comment:
The last arg should be an FP8 recipe:
```suggestion
    return recipe.scaling_factor_compute_algo(amax, scale, fp8_max, recipe)
```
See: https://github.com/NVIDIA/TransformerEngine/blob/08e5e4b1be2e3eea40d4dabda0b62cd4d60d6fab/transformer_engine/common/recipe/__init__.py#L85-L97

That said, I don't see this API used in NeMo or Megatron-LM. Currently the only usage is in a test:
https://github.com/NVIDIA/TransformerEngine/blob/08e5e4b1be2e3eea40d4dabda0b62cd4d60d6fab/tests/pytorch/test_sanity.py#L96-L102
It probably wouldn't be too disruptive to change the API as long as you make the corresponding change in `test_sanity.py`. However, I think this would be a bad idea since it commits us to delayed scaling recipes, while the user (or us in the future) may want to do something more exotic using other attributes in the recipe. Pinging @ksivaman @ptrendx.